app/models/wpcc/contributions_calendar.rb

```diff
@@ -0,0 +1,56 @@
+module Wpcc
+  class ContributionsCalendar
+    attr_reader :eligible_salary,
+                :salary_frequency,
+                :employee_percent,
+                :employer_percent
+
+    PERIODS_FILE = Wpcc::Engine.root.join('config', 'contribution_periods.yml')
+    PERIODS = YAML.load_file(PERIODS_FILE)
+
+    def initialize(eligible_salary:,
+                   employee_percent:,
+                   employer_percent:,
+                   salary_frequency:)
+
+      @eligible_salary = eligible_salary
+      @employee_percent = employee_percent
+      @employer_percent = employer_percent
+      @salary_frequency = salary_frequency
+    end
+
+    def schedule
```

Comment:
Yes, this needs a conversation between us in person.

Because the way that I was thinking is for the application code does not know anything about the periods that will be displayed. This will be more on the yaml  file.

Then on the view you can do something like:

```
  @schedule.each do |period|
      period.employee_percent
      period.employer_percent
      period.tax_relief
      period.total_contributions
   end
```

Then you don't even touch on the periods itself and it will work fine.